finetune/adapter.py

```diff
@@ -85,19 +85,19 @@ def setup(
         main,
         devices,
         Config.from_name(name=checkpoint_dir.name),
-        IOArgs(data_dir, checkpoint_dir, out_dir),
+        IOArgs(data_dir=data_dir, checkpoint_dir=checkpoint_dir, out_dir=out_dir),
```

Comment:
Just FIY: if names of expected args match names of provided ones, you can use something like [this](https://github.com/Andrei-Aksionov/nanoGPTplus/blob/c2eedfe406bf7c8851c393825c09280a7e329242/src/utils/arguments.py#L7) function. 
(Maybe the name isn't the best.)

So it can be something like:
```python

hparams = locals()
...
IOArgs(**grab_arguments(hparams)),
TrainArgs(**grab_arguments(hparams)),
...
```

Reply:
Too magic. And there are arguments named `train_epoch_size` in the hparams but `epoch_size` in the training args

Reply:
Yep. Agree.
But now you know one more magic spell 😉